smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/protocols/AwsProtocolTrait.java

```diff
@@ -80,6 +80,18 @@ protected Node createNode() {
         return builder.build();
     }
 
+    /**
+     * By default, service-side implementation of AWS Protocols send the error shape ID on the wire,
+     * and clients can use it to resolve the error type.
+     *
+     * However, this method can be overridden by legacy protocols implementation which only sends the shape name.
+     *
+     * @return a boolean indicating whether protocol sends fully qualified error shape ID on the wire
+     */
+    public boolean sendErrorShapeNameOverWire() {
```

Comment:
Can we just hardcode the list of protocols that don’t support this? That way we can refactor this however we like in the future.